radix-engine/src/vm/wasm/constants.rs

```diff
@@ -1,31 +1,32 @@
 //=================
-// WASM Shim
+// Blueprint
 //=================
-pub const CONSUME_BUFFER_FUNCTION_NAME: &str = "consume_buffer";
+pub const BLUEPRINT_CALL_FUNCTION_NAME: &str = "blueprint_call";
 
 //=================
-// Costing
+// Address
+//=================
+pub const ADDRESS_ALLOCATE_FUNCTION_NAME: &str = "address_allocate";
+pub const ADDRESS_GET_RESERVATION_ADDRESS_FUNCTION_NAME: &str = "address_get_reservation_address";
+
 //=================
-pub const CONSUME_WASM_EXECUTION_UNITS_FUNCTION_NAME: &str = "gas";
-pub const EXECUTION_COST_UNIT_LIMIT_FUNCTION_NAME: &str = "execution_cost_unit_limit";
-pub const EXECUTION_COST_UNIT_PRICE_FUNCTION_NAME: &str = "execution_cost_unit_price";
-pub const FINALIZATION_COST_UNIT_LIMIT_FUNCTION_NAME: &str = "finalization_cost_unit_limit";
-pub const FINALIZATION_COST_UNIT_PRICE_FUNCTION_NAME: &str = "finalization_cost_unit_price";
-pub const TIP_PERCENTAGE_FUNCTION_NAME: &str = "tip_percentage";
-pub const FEE_BALANCE_FUNCTION_NAME: &str = "fee_balance";
+// Object
+//=================
+pub const OBJECT_NEW_FUNCTION_NAME: &str = "object_new";
+pub const OBJECT_GLOBALIZE_FUNCTION_NAME: &str = "object_globalize";
+pub const OBJECT_GET_BLUEPRINT_ID_FUNCTION_NAME: &str = "object_get_blueprint_id";
+pub const OBJECT_GET_OUTER_OBJECT_FUNCTION_NAME: &str = "object_get_outer_object";
+pub const OBJECT_CALL_FUNCTION_NAME: &str = "object_call";
+pub const OBJECT_CALL_DIRECT_FUNCTION_NAME: &str = "object_call_direct";
+pub const OBJECT_CALL_MODULE_FUNCTION_NAME: &str = "object_call_module";
```

Comment:
While I like most of "entity" prefix to the APIs, I prefer to have `invocation` as a dedicated category. 

Object is more the receiver of an invocation, rather than the definer. "calls" are defined on the system.

It could be something like:
```
call_method(node_id, ident)
call_module_method(node_id, module_id, ident)
call_direct_access_method(node_id, ident)
call_function(blueprint_id, ident)
```

Reply:
I guess this is mostly matter of preference/frame of reference. I had it that way originally but found this version to have nice properties such as "if I have an object, what can I do with it", "if i have a blueprint, what can i do with it". 

That is, all of the APIs use objects as frames of reference rather than functions (ie object oriented).

Lets chat around this in our 1:1

Reply:
Yep!